packages/core/src/Popout/Popout.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { RefObject, useCallback, useEffect, useLayoutEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import { Box } from '..'
+
+export interface IPopoutProps {
+  trigger: JSX.Element
+  content?: JSX.Element | undefined
+  triggerRef?: RefObject<HTMLElement>
+  onOpen?: () => void
+  onClose?: () => void
+  closeOnTriggerRefClick?: boolean
+}
+
+const TriggerWrapper = styled(Box)`
+  position: relative;
```

Comment:
NB: we have a `Relative` component that should work here. You might even be able to set `width='fit-content'` as a prop. although I haven't tried that before.